docs/content/kafka-authorization.md

```diff
@@ -25,87 +25,114 @@ rules to reuse logic and improve overall readability.
+one or the OPA authorizer plugin.
 
 ```bash
-mkdir -p policies
+mkdir policies bundles plugin
 ```
 
+Next, create an OPA policy that allows all requests. You will update this policy later in the tutorial.
+
 **policies/tutorial.rego**:
 
 ```live:start:module:read_only
 package kafka.authz
 
-allow = true
+allow := true
+```
+
+With the policy in place, build a bundle from the contents of the `policies` directory and place it in the `bundles`
+directory. The `bundles` directory will later be mounted into the Nginx container in order to distribute policy updates
+to OPA.
+
+```shell
+opa build --bundle policies/ --output bundles/bundle.tar.gz
+```
+
+#### Kafka Authorizer JAR File
+
+Next, download the latest version of the [Open Policy Agent plugin for Kafka authorization](https://github.com/anderseknert/opa-kafka-plugin)
+plugin from the projects [release pages](https://github.com/anderseknert/opa-kafka-plugin/releases).
+
+Store the plugin in the `plugin` directory (replace `${version}` with the version number of the plugin just downloaded):
+```bash
+mv opa-authorizer-${version}-all.jar plugin/
 ```
 
-Next, create a `docker-compose.yaml` file that runs OPA, ZooKeeper, and Kafka.
+For more information on how to configure the OPA plugin for Kafka, see the plugin [repository](https://github.com/anderseknert/opa-kafka-plugin).
+
+Next, create a `docker-compose.yaml` file that runs OPA, Nginx, ZooKeeper, and Kafka.
 
 **docker-compose.yaml**:
 
 ```yaml
-version: "2"
 services:
+  nginx:
+    image: nginx:1.21.4
+    volumes:
+      - "./bundles:/usr/share/nginx/html"
+    ports:
+      - "80:80"
   opa:
-    hostname: opa
-    image: openpolicyagent/opa:{{< current_docker_version >}}
+    image: openpolicyagent/opa:{{< current_docker_version >}}-rootless
     ports:
-      - 8181:8181
-    # WARNING: OPA is NOT running with an authorization policy configured. This
-    # means that clients can read and write policies in OPA. If you are deploying
-    # OPA in an insecure environment, you should configure authentication and
-    # authorization on the daemon. See the Security page for details:
-    # https://www.openpolicyagent.org/docs/security.html.
-    command: "run --server --watch /policies"
-    volumes:
-      - ./policies:/policies
+      - "8181:8181"
+    command:
+      - "run"
+      - "--server"
+      - "--set=decision_logs.console=true"
+      - "--set=services.authz.url=http://nginx"
+      - "--set=bundles.authz.service=authz"
+      - "--set=bundles.authz.resource=bundle.tar.gz"
+    depends_on:
+      - nginx
   zookeeper:
-    image: confluentinc/cp-zookeeper:4.0.0-3
+    image: confluentinc/cp-zookeeper:6.2.1
+    ports:
+      - "2181:2181"
     environment:
-      ZOOKEEPER_CLIENT_PORT: 2181
-      zk_id: "1"
-  kafka:
-    hostname: kafka
-    image: openpolicyagent/demo-kafka:1.0
-    links:
-      - zookeeper
-      - opa
+      - ALLOW_ANONYMOUS_LOGIN=yes
+      - ZOOKEEPER_CLIENT_PORT=2181
+  broker:
```

Comment:
The `broker` service is missing an image

Reply:
Ouch! Very good catch 😬 Looks like this was removed with the comment above, since this was all really flakey on my M1. 

```yaml
    # If experiencing hangs on darwin/arm64, explicitly setting the platform here seems to help
    # platform: linux/amd64
    image: confluentinc/cp-kafka:6.2.1
```

Adding the image back!